So this fails on https://github.com/ansible/ansible/blob/6410570643256d828dce52cd3935c1502102048e/test/integration/targets/inventory_aws_conformance/inventory_diff.py#L36 which is not in devel.

```
Traceback (most recent call last):
  File "./inventory_diff.py", line 66, in <module>
    main()
  File "./inventory_diff.py", line 62, in main
    check_host_vars(key, value, bdata, ahost)
  File "./inventory_diff.py", line 38, in check_host_vars
    assert value == plugin['_meta']['hostvars'][host][key], "%s != %s" % (value, plugin['_meta']['hostvars'][host][key])
AssertionError: 123456789012 != 123456789012
```

The values that are being compared are of different types:
```
value=123456789012 of type=<class 'str'>
plugin['_meta']['hostvars'][host][key]=123456789012 of type=<class 'int'>
```

I am looking into why is that happening.

--- a/test/integration/targets/inventory_aws_conformance/runme.sh
+++ b/test/integration/targets/inventory_aws_conformance/runme.sh
@@ -83,7 +83,7 @@ compose:
   ec2_kernel: kernel_id | default("")
   ec2_monitored:  monitoring['state'] in ['enabled', 'pending']
   ec2_monitoring_state: monitoring['state']
-  ec2_account_id: owner_id
+  ec2_account_id: owner_id | string | regex_replace('^(.*)$', '"\\1"')  # work around the fact that the string filter is not reliable on native jinja
   ec2_placement: placement['availability_zone']
   ec2_ramdisk: ramdisk_id | default("")
   ec2_reason: state_transition_reason
@@ -95,7 +95,7 @@ compose:
   ec2_sourceDestCheck: source_dest_check | lower | string  # butchered snake_case case not a typo.
 
   # vars that just need ec2_ prefix
-  ec2_ami_launch_index: ami_launch_index | string
+  ec2_ami_launch_index: ami_launch_index | string | regex_replace('^(.*)$', '"\\1"')  # see ec2_account_id for why this needs to be quoted
   ec2_architecture: architecture
   ec2_client_token: client_token
   ec2_ebs_optimized: ebs_optimized

